stock_indicators/_cstypes/decimal.py

```diff
@@ -33,4 +34,4 @@ def to_pydecimal(cs_decimal):
         cs_decimal : `System.Decimal` of C# or any `object` that can be represented as a number.
     """
     if cs_decimal is not None:
-        return PyDecimal(str(cs_decimal))
+        return PyDecimal(cs_decimal.ToString(CsCultureInfo.InvariantCulture))
```

Comment:
Does this still need to double convert to/from a string?  It’d be nice if it were more direct, like:

```python
return PyDecimal(cs_decimal, CsCultureInfo.InvariantCulture)
```

Reply:
Yeap. We need that for:
 - Keep its precision.
 - Convert it with InvariantCulture.


PyDecimal is actually built-in Decimal class and it does not have constructor passing `CsCultureInfo` as a second parameter.

Reply:
I see there are [others who are struggling](https://github.com/pythonnet/pythonnet/issues/2319) with this expensive requirements to double convert with string values.

Reply:
I had a bit of [a longer conversation with Chad]([url](https://chatgpt.com/share/eedd6957-a5de-4cd4-a3a7-e2411269f854)) about this.  He's saying the only way to avoid using string conversion is to simply use the .NET decimal type in Python (not trying to convert to Python decimal).  Is that feasible in our use case?

Reply:
There's nothing C# Decimal itself can not be used in Python "technically". The point is users have to know the C# Decimal interface when they use or manipulate the result values in their own logic or somewhat reason. That's the wrapper lost its purpose.

Reply:
Here's an another history I asked on StackOverflow at very early time. 
https://stackoverflow.com/questions/66841211/convert-python-floating-point-into-c-sharp-decimal

Reply:
I'd suggest trying the double hop through float (or double) instead of string, but I can see that the entire `locale` issue is tied to the `string` representation, right?  Ugh.

Reply:
Since this string conversion is only needed when user's are supplying their own data _when loading quotes_, are we also using it during normal back and forth with the library and for return values?  If so, maybe we just try to limit its impact on Quote inserts from users so normal calculations don't need to suffer the performance cost of string conversions?

> We may already be doing this; I haven't dug through the code to see when/where `string` conversions are being used.

Reply:
It is only used for getter method of Decimal property, which means that it is only called(convert into PyDecimal using string) at the moment users access the decimal property. Nothing else.
So I think it does not affect on the performance. Of course, if we include accessing the result value in performance test, we can say it affects on the our performance test. But inevitable. When the users try to get/access the result value, I'm sure definitely they don't want to deal with C# object in their Python code.

Reply:
Yeap it is neccessary to convert back into Python Decimal. I will test the snippets you suggest works

Reply:
Generally, rule of thumb, converting numeric types to/from `string` are going to be noticeably slower than converting between two numeric types, given the way the have to write to memory.

Reply:
Ok.  I guess we’ll just monitor Python.NET over time to see if they can come up with a better built-in conversion codec for `decimal`.